rebasehelper/specfile.py

```diff
@@ -23,35 +23,58 @@
 #          František Nečas <fifinecas@seznam.cz>
 
 import argparse
-import collections
 import enum
 import itertools
-import locale
 import logging
 import os
 import re
 import shlex
 import shutil
-import urllib.parse
-from datetime import date
-from difflib import SequenceMatcher
-from operator import itemgetter
-from typing import List, Optional, Pattern, Tuple, Dict, Union, cast
+from typing import Callable, List, Optional, Pattern, Tuple, Dict, cast
 
-import rpm  # type: ignore
+from specfile import Specfile
+from specfile.exceptions import RPMException
+from specfile.macros import MacroLevel
+from specfile.sections import Section
+from specfile.sources import Source, ListSource, TagSource
 
 from rebasehelper import constants
 from rebasehelper.archive import Archive
-from rebasehelper.spec_content import SpecContent
-from rebasehelper.tags import Tag, Tags
 from rebasehelper.exceptions import RebaseHelperError, DownloadError, ParseError, LookasideCacheError
 from rebasehelper.argument_parser import SilentArgumentParser
 from rebasehelper.logger import CustomLogger
 from rebasehelper.helpers.download_helper import DownloadHelper
-from rebasehelper.helpers.macro_helper import MacroHelper
-from rebasehelper.helpers.rpm_helper import RpmHelper, RpmHeader
 from rebasehelper.helpers.git_helper import GitHelper
 from rebasehelper.helpers.lookaside_cache_helper import LookasideCacheHelper
+from rebasehelper.helpers.rpm_helper import RpmHeader
+
+
+MACROS_WHITELIST: List[str] = [
+    '_bindir',
+    '_datadir',
+    '_includedir',
+    '_infodir',
+    '_initdir',
+    '_libdir',
+    '_libexecdir',
+    '_localstatedir',
+    '_mandir',
+    '_sbindir',
+    '_sharedstatedir',
+    '_sysconfdir',
+    'python2_sitelib',
+    'python3_sitelib',
+]
+
+# Comments in these sections can only be on a separate line.
+DISALLOW_INLINE_COMMENTS: List[str] = [
```

Comment:
I am wondering if this shouldn't somehow be included in specfile since it deals with specfile parsing but looks like it's not needed in the library itself?

Reply:
The thing is, comments in spec files can never be inline. But sections such as `%prep`, `%build` etc. actually contain shell scripts, where inline comments are valid.
Perhaps it would make sense if `Section` had something like a `shell` attibute, and then we could use that rather than this constant?

Reply:
https://github.com/packit/specfile/pull/170